### PR TITLE
Don't disconnect the wallet and retry change the chain on MM error with code 1013

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

#923

There is a patch for `@web3-react/metamask` package that ignores  the MetaMask error with code 1013 and retries to change network.

**The patch should be replaced by the fix: https://github.com/Uniswap/web3-react/pull/653**

  # To Test
See #923

  # Background
The problem can be reproduced by the script:
```
window.postMessage({
    "target": "metamask-inpage",
    "data": {
        "name": "metamask-provider",
        "data": {
            "method": "metamask_chainChanged",
            "params": {
                "chainId": "0x64",
                "networkVersion": "loading"
            }
        }
    }
}, '*')
```

After running this script we get the wallet disconnection.
It's related to the MetaMask `inpage.js` code:
```
 _handleChainChanged({chainId: e, networkVersion: t}={}) {
    l.isValidChainId(e) && l.isValidNetworkVersion(t) ? "loading" === t ? this._handleDisconnect(!0) : super._handleChainChanged({
        chainId: e
    }) : this._log.error(u.default.errors.invalidNetworkParams(), {
        chainId: e,
        networkVersion: t
    })
}
```

In case if the `networkVersion` parameter has 'loading' as value they just disconnect the wallet.

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @shoom3301 , changes LGTM!
The only thing that I noticed is that  the app shows 'connect to wallet' button for 1-2 sec when change a network to GC now. But then it is reconnected. 
I opened a separate issue #925 to address it later. 

Thanks!

**Review by elena-zh — APPROVED** [review]

**Review by nenadV91 — APPROVED** [review]
I really like the `patch-package` approach. 
Codewise LGTM,

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -88,9 +88,9 @@ jobs:
         uses: actions/cache@v2
         with:
           path: node_modules
-          key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules1-${{ hashFiles('**/yarn.lock') }}
```
Comment (anxolin) [inline .github/workflows/ci.yml:91]: were this changes to `nodemodules1` intended?

**Review by anxolin — COMMENTED** [review]
I love the `patch-package`. I was not aware of that, and I usually deal with this by using a modified version of the package until the PR is merged. This is so much better :) 

Only concerned is why we changed the cache. I guess now is tested we can revert this part? 
BTW, also there's an API: https://docs.github.com/en/rest/actions/cache

**Pull request merged**